### ISSUE TITLE
Basemap without internet

### ISSUE THREAD

**Issue by GoogleCodeExporter** [lixun910/geoda#208]
```
GeoDa 1.7.45

Ran into this in Win 7: Without an internet connection, when you add basemap, 
no basemap is added but program slows down (e.g. zoom in and out: not 
responding - doesn't crash but longer delays). 

Can we detect in advance if internet connection exists and only enable basemap 
option when it does? 
Or some other option to avoid slower performance with basemap artificially 
active (since it's not displaying) without internet?
```

Original issue reported on code.google.com by `jkoschin...@gmail.com` on 27 Jul 2015 at 12:37

**Comment by lixun910** [comment]
fix is proposed in 276b615 and 8d0b20e.

main change is CURLOPT_CONNECTTIMEOUT is set to 1 second, so that the response from curlib is faster.

**Comment by lixun910** [comment]
fix is applied in 1.7.49

**Comment by jkoschinsky** [comment]
Verified in GeoDa 1.7.63
